src/udev/scsi_id/scsi_serial.c

```diff
@@ -612,7 +612,7 @@ static int do_scsi_page83_inquiry(struct scsi_id_device *dev_scsi, int fd,
                  * Examine each descriptor returned. There is normally only
                  * one or a small number of descriptors.
                  */
-                for (j = 4; j <= (unsigned)page_83[3] + 3; j += page_83[j + 3] + 4) {
+                for (j = 4; j <= (unsigned)(page_83[2] << 8) + (unsigned)page_83[3] + 3; j += page_83[j + 3] + 4) {
```

Comment:
Hmm. Maybe,
```
(unsigned) page_83[2] << 8 + (unsigned) page_83[3]
```
Otherwise, shifting `unsigned char` with `8` vanishes any value? Not sure.

Reply:
Updated.